### PR TITLE
Ruleset and error changes

### PR DESCRIPTION
**Pull request by cbetta**
* Splits conditional rules into separate rules for comparing numbers, booleans, and strings.
* Renames terminal rules to `payment-services-terminal` as we might have other terminal rules in the future.
* Reformats our errors to take a list of details. This can be used by Pydantic to fill in some extra details about a validation error.

### CODE REVIEW

File: reference/components/card_rule_boolean_conditional.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  key:
+    type: string
+    description: The transaction field to filter by.
+    example: store
+    enum:
+      - store
+      - use_test_environment
+
+  value:
```
Comment (danielknell) [inline reference/components/card_rule_boolean_conditional.yaml:48]: i thought you wanted to move away from `== true` or `== false` condition/value to `is-false` and `is-true` just condition style for bools?

Comment (cbetta) [reply reference/components/card_rule_boolean_conditional.yaml:48]: I removed the conditional, e.g. `==` vs `!=` but we somehow need to set if it should match on true or false....

Comment (cbetta) [reply reference/components/card_rule_boolean_conditional.yaml:48]: And I didn't wan to introduce 2 new fields just for this one....

Comment (cbetta) [reply reference/components/card_rule_boolean_conditional.yaml:48]: You are right though, we dont need either really. Let me update.

File: reference/components/payment_service_definition.yaml [hunk 2]
```diff
@@ -75,6 +75,7 @@ properties:
       A list of three-letter ISO currency codes that this service supports
     items:
       type: string
```
Comment (danielknell) [inline reference/components/payment_service_definition.yaml:77]:
```suggestion
      type: string
      minLength: 3
      maxLength: 3
```

File: reference/components/payment_service_definition.yaml [hunk 3]
```diff
@@ -86,6 +87,7 @@ properties:
       A list of two-letter ISO country codes that this service supports
     items:
       type: string
```
Comment (danielknell) [inline reference/components/payment_service_definition.yaml:89]:
```suggestion
      type: string
      minLength: 2
      maxLength: 2
```

File: reference/components/payment_service_definition_create_request.yaml [hunk 4]
```diff
@@ -55,14 +61,21 @@ properties:
...
           description: The name to display for a field in the cluster UI
+          minLength: 1
+          maxLength: 50
         help_url:
           type: string
+          format: uri
           example: https://stripe.com/docs/private_api_key
           description: A URL that provides more details about this field
+          minLength: 1
+          maxLength: 1000
```
Comment (danielknell) [inline reference/components/payment_service_definition_create_request.yaml:78]:
```suggestion
```
the more I think about it the less it feels like fixed lengths for formats that have a defined upper limit make sense...

File: reference/components/payment_service_definition_create_request.yaml [hunk 5]
```diff
@@ -71,15 +84,21 @@ properties:
 
   help_url:
     type: string
+    format: uri
     example: https://stripe.com/docs
     description: A URL that provides more details about this payment service
+    minLength: 1
+    maxLength: 1000
```
Comment (danielknell) [inline reference/components/payment_service_definition_create_request.yaml:91]:
```suggestion
```

File: reference/components/payment_service_definitions.yaml [hunk 6]
```diff
@@ -25,11 +25,15 @@ properties:
   next_page:
     type: string
     description: The marker that represents the next page of results.
-    example: stripe-card
+    minLength: 1
+    maxLength: 1000
+    example: ZXhhbXBsZTE=
```
Comment (danielknell) [inline reference/components/payment_service_definitions.yaml:30]:
```suggestion
    example: ZXhhbXBsZTE
```

File: reference/components/payment_service_definitions.yaml [hunk 7]
```diff
@@ -25,11 +25,15 @@ properties:
...
     nullable: true
 
   previous_page:
     type: string
+    minLength: 1
+    maxLength: 1000
     description: The marker that represents the previous page of results.
-    example: null
+    example: ZXhhbXBsZTI=
```
Comment (danielknell) [inline reference/components/payment_service_definitions.yaml:38]:
```suggestion
    example: ZXhhbXBsZTI
```

File: reference/components/payment_services.yaml [hunk 8]
```diff
@@ -24,13 +24,15 @@ properties:
   next_page:
     type: string
     description: The marker that represents the next page of results.
-    format: uuid
-    example: 9f944c64-eaf0-4219-90fd-f7e56c7cd214
+    minLength: 1
+    maxLength: 1000
+    example: ZXhhbXBsZTE=
```
Comment (danielknell) [inline reference/components/payment_services.yaml:29]:
```suggestion
    example: ZXhhbXBsZTE
```

File: reference/components/payment_services.yaml [hunk 9]
```diff
@@ -24,13 +24,15 @@ properties:
...
     nullable: true
 
   previous_page:
     type: string
+    minLength: 1
+    maxLength: 1000
     description: The marker that represents the previous page of results.
-    format: uuid
-    example: 9049b2f9-0c67-47c5-af18-e8525f855263
+    example: ZXhhbXBsZTI=
```
Comment (danielknell) [inline reference/components/payment_services.yaml:37]:
```suggestion
    example: ZXhhbXBsZTI
```

File: reference/components/transaction_events.yaml [hunk 10]
```diff
@@ -25,13 +25,15 @@ properties:
   next_page:
     type: string
     description: The marker that represents the next page of results.
-    format: uuid
-    example: 9f944c64-eaf0-4219-90fd-f7e56c7cd214
+    minLength: 1
+    maxLength: 1000
+    example: ZXhhbXBsZTE=
```
Comment (danielknell) [inline reference/components/transaction_events.yaml:30]:
```suggestion
    example: ZXhhbXBsZTE
```

File: reference/components/transaction_events.yaml [hunk 11]
```diff
@@ -25,13 +25,15 @@ properties:
...
     nullable: true
 
   previous_page:
     type: string
+    minLength: 1
+    maxLength: 1000
     description: The marker that represents the previous page of results.
-    format: uuid
-    example: 9049b2f9-0c67-47c5-af18-e8525f855263
+    example: ZXhhbXBsZTI=
```
Comment (danielknell) [inline reference/components/transaction_events.yaml:38]:
```suggestion
    example: ZXhhbXBsZTI
```

File: reference/components/transactions.yaml [hunk 12]
```diff
@@ -25,13 +25,15 @@ properties:
   next_page:
     type: string
     description: The marker that represents the next page of results.
-    format: uuid
-    example: 9f944c64-eaf0-4219-90fd-f7e56c7cd214
+    minLength: 1
+    maxLength: 1000
+    example: ZXhhbXBsZTE=
```
Comment (danielknell) [inline reference/components/transactions.yaml:30]:
```suggestion
    example: ZXhhbXBsZTE
```

File: reference/components/transactions.yaml [hunk 13]
```diff
@@ -25,13 +25,15 @@ properties:
...
     nullable: true
 
   previous_page:
     type: string
+    minLength: 1
+    maxLength: 1000
     description: The marker that represents the previous page of results.
-    format: uuid
-    example: 9049b2f9-0c67-47c5-af18-e8525f855263
+    example: ZXhhbXBsZTI=
```
Comment (danielknell) [inline reference/components/transactions.yaml:38]:
```suggestion
    example: ZXhhbXBsZTI
```

File: reference/paths/buyers__list-buyers.yaml [hunk 14]
```diff
@@ -28,7 +28,7 @@ parameters:
   - schema:
       type: string
       format: ""
-      example: "1593343179"
+      example: "ZXhhbXBsZTE="
```
Comment (danielknell) [inline reference/paths/buyers__list-buyers.yaml:31]:
```suggestion
      example: "ZXhhbXBsZTE"
```

File: reference/paths/card_rulesets__list-card-rulesets.yaml [hunk 15]
```diff
@@ -20,7 +20,7 @@ parameters:
   - schema:
       type: string
       format: ""
-      example: "1593343179"
+      example: "ZXhhbXBsZTE="
```
Comment (danielknell) [inline reference/paths/card_rulesets__list-card-rulesets.yaml:23]:
```suggestion
      example: "ZXhhbXBsZTE"
```

File: reference/paths/payment_methods__list-payment-method-events.yaml [hunk 16]
```diff
@@ -20,7 +20,7 @@ parameters:
   - schema:
       type: string
       format: ""
-      example: "1593343179"
+      example: "ZXhhbXBsZTE="
```
Comment (danielknell) [inline reference/paths/payment_methods__list-payment-method-events.yaml:23]:
```suggestion
      example: "ZXhhbXBsZTE"
```

File: reference/paths/payment_methods__list-payment-methods.yaml [hunk 17]
```diff
@@ -20,7 +20,7 @@ parameters:
   - schema:
       type: string
       format: ""
-      example: "1593343179"
+      example: "ZXhhbXBsZTE="
```
Comment (danielknell) [inline reference/paths/payment_methods__list-payment-methods.yaml:23]:
```suggestion
      example: "ZXhhbXBsZTE"
```

File: reference/paths/payment_service_definitions__list-payment-service-definitions.yaml [hunk 18]
```diff
@@ -20,7 +20,7 @@ parameters:
   - schema:
       type: string
       format: ""
-      example: "1593343179"
+      example: "ZXhhbXBsZTE="
```
Comment (danielknell) [inline reference/paths/payment_service_definitions__list-payment-service-definitions.yaml:23]:
```suggestion
      example: "ZXhhbXBsZTE"
```

File: reference/paths/payment_services__list-payment-services.yaml [hunk 19]
```diff
@@ -22,7 +22,7 @@ parameters:
   - schema:
       type: string
       format: ""
-      example: "1593343179"
+      example: "ZXhhbXBsZTE="
```
Comment (danielknell) [inline reference/paths/payment_services__list-payment-services.yaml:25]:
```suggestion
      example: "ZXhhbXBsZTE"
```

File: reference/paths/transactions__list-transaction-events.yaml [hunk 20]
```diff
@@ -20,7 +20,7 @@ parameters:
   - schema:
       type: string
       format: ""
-      example: "1593343179"
+      example: "ZXhhbXBsZTE="
```
Comment (danielknell) [inline reference/paths/transactions__list-transaction-events.yaml:23]:
```suggestion
      example: "ZXhhbXBsZTE"
```

File: reference/paths/transactions__list-transactions.yaml [hunk 21]
```diff
@@ -19,7 +19,7 @@ parameters:
 
   - schema:
       type: string
-      example: "1593343179"
+      example: "ZXhhbXBsZTE="
```
Comment (danielknell) [inline reference/paths/transactions__list-transactions.yaml:22]:
```suggestion
      example: "ZXhhbXBsZTE"
```

File: reference/components/payment_service.yaml [hunk 22]
```diff
@@ -24,6 +26,8 @@ properties:
     description: |
       The ID of the payment service used to create this service instance.
     example: stripe
```
Comment (danielknell) [inline reference/components/payment_service.yaml:28]:
```suggestion
    example: stripe-card
```

File: reference/components/payment_service.yaml [hunk 23]
```diff
@@ -24,6 +26,8 @@ properties:
     description: |
       The ID of the payment service used to create this service instance.
```
Comment (danielknell) [inline reference/components/payment_service.yaml:27]: this is a payment service but a payment service was used to create it?

File: reference/components/payment_service_update_request.yaml [hunk 24]
```diff
@@ -51,6 +51,7 @@ properties:
       two-letter code format.
     items:
       type: string
```
Comment (danielknell) [inline reference/components/payment_service_update_request.yaml:53]:
```suggestion
      type: string
      minLength: 2
      maxLength: 2
```

also needed for currencies above.

File: reference/components/payment_service_create_request.yaml [hunk 25]
```diff
@@ -20,6 +20,8 @@ properties:
     description:
       A custom name for the payment service. This will be shown in the Admin UI.
     example: Stripe (USA)
+    minLength: 1
+    maxLength: 50
 
   payment_service_definition_id:
     type: string
```
Comment (danielknell) [inline reference/components/payment_service_create_request.yaml:27]: missing min/max length, should probably be `stripe-card` and if we are creating a payment service we need to use a "payment service"?

File: reference/components/payment_service_create_request.yaml [hunk 26]
```diff
@@ -47,6 +49,8 @@ properties:
           type: string
```
Comment (danielknell) [inline reference/components/payment_service_create_request.yaml:49]: add maxLength 50 to key for consistency?

File: reference/components/payment_service.yaml (hunk 23 above)
Comment (cbetta) [reply reference/components/payment_service.yaml:27]:
```suggestion
      The ID of the payment service definition used to create this service.
```

**Pull request merged**